mongo/collection_test.go

```diff
@@ -309,8 +309,14 @@ func TestNewFindOptionsFromFindOneOptions(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			t.Parallel()
 
-			got := newFindOptionsFromFindOneOptions(test.opts...)
-			assert.Equal(t, test.want, got)
+			var err error
+			got := &options.FindArgs{}
+			err = options.Merge(got, []options.Options[options.FindArgs]{newFindOptionsFromFindOneOptions(test.opts...)}...)
```

Comment:
We can simplify the syntax here by providing an explicit type for `options.Merge`.

```suggestion
			err = options.Merge[options.FindArgs](got, newFindOptionsFromFindOneOptions(test.opts...))
```